File: src/main/java/no/knowit/spring/springtutorial/data/Melding.java
package no.knowit.spring.springtutorial.data;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.NonNull;

import javax.persistence.*;

@Getter
@NoArgsConstructor
@Entity
public class Melding {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String json;
    @Enumerated(EnumType.STRING)
    private Type type;

    public Melding(@NonNull String json, @NonNull Type type) {
        this.json = json;
        this.type = type;
    }

    public boolean hasId() {
        return id > 0;
    }
}

File: src/main/java/no/knowit/spring/springtutorial/data/Melding.kt
package no.knowit.spring.springtutorial.data

import lombok.NonNull
import javax.persistence.*

@Entity
class Melding(@NonNull val json: String, @NonNull @Enumerated(EnumType.STRING) val type: Type) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null

    fun hasId(): Boolean {
        return id != null
    }
}
